crates/ruff/src/rules/pycodestyle/rules/whitespace_before_parameters.rs

```diff
@@ -30,27 +31,31 @@ impl AlwaysAutofixableViolation for WhitespaceBeforeParameters {
 
 /// E211
 #[cfg(feature = "logical_lines")]
-pub fn whitespace_before_parameters(
-    tokens: &[(Location, &Tok, Location)],
-    autofix: bool,
-) -> Vec<Diagnostic> {
+pub fn whitespace_before_parameters(tokens: &LogicalLineTokens, autofix: bool) -> Vec<Diagnostic> {
     let mut diagnostics = vec![];
-    let (_, mut prev_token, mut prev_end) = tokens.first().unwrap();
-    for (idx, (start, tok, end)) in tokens.iter().enumerate() {
-        if is_op_token(tok)
-            && (**tok == Tok::Lpar || **tok == Tok::Lsqb)
```

Comment:
I starred at this code for a long time and have come to the conclusion that checking `is_op_token(tok)` isn't necessary because we narrow down the token to `Lpar` or `Lsqb` on the very next line...